### PR TITLE
[GEAR-943] ENH - Readers group and readers progress report

### PR DESCRIPTION
**Pull request by dpark6060**
Readers group is now the same group as the main project
Readers progress report is now generated with every gather cases run, indicating progress of current assigned cases and max cases
Fixed issues with list comprehension and flywheel permissions
In Readers projects, there can only be ONE user with "read-write" or "read-only" permissions, and that user MUST be the reader.  All other users can be admins or other roles.  Keep this in mind when using group-level permission templates.  

In my mind this is a problem that will hurt us later.  But hey, I just do what I'm told.

### CODE REVIEW

**Comment by AndysWorth** [comment]
Assuming this is being used by customers and they are not complaining, LGTM.

File: gears/assign_batch_cases/utils/manage_cases.py [hunk 1]
```diff
@@ -0,0 +1,621 @@
...
+
+    group_projects = fw_client.projects.find(f'group="{group_id}"')
+
+    proj_roles = [
+        role.id
+        for role in fw_client.get_all_roles()
+        if role.label in ["read-write", "read-only"]
+    ]
+
+    # valid_reader_ids = [
```
Comment (neurosutton) [inline gears/assign_batch_cases/utils/manage_cases.py:209]: Do you want to clean up definitions that are not used anymore?

File: gears/assign_batch_cases/utils/manage_cases.py [hunk 2]
```diff
@@ -0,0 +1,621 @@
...
+        src_session = fw_client.sessions.find_first(f"_id={session_id}")
+
+        if src_session:
+            src_session = src_session.reload()
+            session_features = set_session_features(src_session, case_coverage)
+        else:
+            session_features = {}
+
+        # Locate Reader Project
+        if reader_email in dest_projects_df.reader_id.values:
```
Comment (neurosutton) [inline gears/assign_batch_cases/utils/manage_cases.py:510]: Regarding your "more graceful way" in line 516, would it help to try:
``` python
project_id = dest_projects_df.loc[dest_projects_df['reader.id'] == reader_email, 'project_id']
```
and ditch line 511?

File: gears/assign_cases/run.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        verify_user_permissions(fw_client, context)
+        check_for_duplicate_execution(fw_client)
+
+        destination_id = context.destination["id"]
+        analysis = fw_client.get(destination_id)
+        source_project = fw_client.get(analysis.parents["project"])
+        reader_group_id = source_project.group
+
+
+        # TODO: Verify that this isn't fucking something up.
```
Comment (neurosutton) [inline gears/assign_cases/run.py:36]: Probably need to remove language for open resource.

**Review by neurosutton — CHANGES_REQUESTED** [review]
Please review commented line 33 in gears/assign_cases/run.py and see conversation comments for a couple of minor suggestions. Otherwise, LGTM.

File: gears/assign_batch_cases/utils/manage_cases.py (hunk 1 above)
Comment (dpark6060) [reply gears/assign_batch_cases/utils/manage_cases.py:209]: commented definitions are the "original".  I had to expand the list comprehension to address an issue, and I want to leave them in until it's been vetted with a bunch of users, so I can always reference the original function incase there's something weird that gets messed up.

File: gears/assign_cases/run.py (hunk 3 above)
Comment (dpark6060) [reply gears/assign_cases/run.py:36]: FINE

Comment (AndysWorth) [reply gears/assign_cases/run.py:36]: git blame knows all, tells all

Comment (dpark6060) [reply gears/assign_cases/run.py:36]: # TODO: Verify that this isn't RUSTLING ANYONE'S JIMMIES

File: gears/assign_batch_cases/utils/manage_cases.py (hunk 2 above)
Comment (dpark6060) [reply gears/assign_batch_cases/utils/manage_cases.py:510]: slight modifications, but done!  Thanks for the help.

File: gears/assign_cases/run.py (hunk 3 above)
Comment (joshicola) [reply gears/assign_cases/run.py]: hahahahahahahahahahahahahahahahahahahahahahaha!!!

**Pull request merged**